routers/api/packages/api.go

```diff
@@ -81,15 +81,17 @@ func Routes() *web.Route {
 						r.Get("/digest", conan.RecipeDownloadURLs)
 						r.Post("/upload_urls", reqPackageAccess(perm.AccessModeWrite), conan.RecipeUploadURLs)
 						r.Get("/download_urls", conan.RecipeDownloadURLs)
-						r.Group("/packages", func() {
-							r.Post("/delete", reqPackageAccess(perm.AccessModeWrite), conan.DeletePackageV1)
-							r.Group("/{package_reference}", func() {
-								r.Get("", conan.PackageSnapshot)
-								r.Get("/digest", conan.PackageDownloadURLs)
-								r.Post("/upload_urls", reqPackageAccess(perm.AccessModeWrite), conan.PackageUploadURLs)
-								r.Get("/download_urls", conan.PackageDownloadURLs)
+						if setting.Packages.Enabled {
```

Comment:
I am absolutely not a friend of offering **API** routes conditionally.
Because then, an app like `GitNex` has no option but to "try-catch" whether a given route is supported for an unknown server.
I think a better alternative would be to make the corresponding functions no-ops in that case, i.e. by simply returning the status `501 - Not Implemented`.

Reply:
I'm uncertain what is a good approach. We already use conditional routes

https://github.com/go-gitea/gitea/blob/9c349a4277926bfd3ff0360301765ad7abd9f10b/routers/api/v1/api.go#L632-L636

https://github.com/go-gitea/gitea/blob/9c349a4277926bfd3ff0360301765ad7abd9f10b/routers/api/v1/api.go#L638-L640

https://github.com/go-gitea/gitea/blob/9c349a4277926bfd3ff0360301765ad7abd9f10b/routers/api/v1/api.go#L669-L671

And all the `mustXXX` methods return a 404 which cannot be distinguished from real not found paths.
https://github.com/go-gitea/gitea/blob/9c349a4277926bfd3ff0360301765ad7abd9f10b/routers/api/v1/api.go#L475